### ISSUE TITLE
Impossible to provide custom issue code

### ISSUE THREAD

**Issue by dilame** [colinhacks/zod#1627]
In your [documentation](https://zod.dev/ERROR_HANDLING?id=zodissuecode) said

> This is the error code throw by refinements (unless you are using `superRefine` in which case it's possible to throw issues of any `code`)

But when i provide custom code TS complains
```
TS2322: Type '"duplicated_id"' is not assignable to type '"invalid_enum_value" | "invalid_type" | "unrecognized_keys" | "invalid_literal" | "invalid_union" | "invalid_union_discriminator" | "invalid_arguments" | "invalid_return_type" | ... 6 more ... | "too_small"'.```

**Comment by colinhacks** [comment]
Yeah the code still needs to be defined in `ZodIssueCode`. This is necessary for the discriminated union to work properly (`ZodIssue` is a big discriminated union of every issue type) as well as Zod's whole error message generation system. I'll think about ways to make this more extensible but I recommend sticking with `"custom"` for now.

**Comment by JacobWeisenburger** [comment]
@dilame, was that a satisfactory answer? Or do you have any other questions? If not, I'd like to close this issue.

**Comment by dilame** [comment]
The answer is ok, but it is not a question but issue, and it still persists – in docs you have written in two places that user can provide custom error code, but actually you can not. I think in order to resolve this issue we need either fix docs or code.

**Comment by JacobWeisenburger** [comment]
I'll take a look at the docs. Would you be able to give me links to where the docs say this?

**Comment by dilame** [comment]
You can use global search to find this text in the main docs page [here](https://zod.dev/?id=superrefine)

> Normally refinements always create issues with a ZodIssueCode.custom error code, **but with superRefine you can create any issue of any code**.

And [here](https://zod.dev/ERROR_HANDLING?id=zodissuecode) (`ZodIssueCode.custom` description)

> unless you are using superRefine in which case it's possible to throw issues of any code

**Comment by dilame** [comment]
And, by the way, my vote is for fixing the code instead of docs because actually i use this feature with `// @ts-ignore` flag.

**Comment by JacobWeisenburger** [comment]
I'm not opposed to changing the code. I just want to make sure the docs are reflecting the latest release.

As far as changing the code, that's something you will need to convince @colinhacks. I'll leave this issue open for you to do that if you want.

**Comment by JacobWeisenburger** [comment]
I updated the docs here: https://github.com/colinhacks/zod/commit/1115bd143744aa903ae9426ac3c116790dc14154
